### PR TITLE
Add On Weapon Created Hook

### PR DESCRIPTION
**Pull request by TRBlount**
This hook is distinct from `$On Weapon Fired:` in that it triggers each time `weapon_create` is called by the engine. Additionally, the weapon object itself is included as a hookvar `Weapon`.

### CODE REVIEW

**Comment by AxemP** [comment]
I have done some basic tests with this and it works as expected.

File: code/scripting/scripting.cpp [hunk 1]
```diff
@@ -315,6 +316,9 @@ bool ConditionedHook::ConditionsValid(int action, object *objp, int more_data)
 							return false;
 					} else if(objp == NULL || objp->type != OBJ_SHIP) {
 						return false;
+					} else if (action == CHA_ONWEAPONCREATED) {
+						if (objp == NULL || objp->type != OBJ_WEAPON)
```
Comment (niffiwan) [inline code/scripting/scripting.cpp]: Travis wants this to be nullptr; not NULL

**Comment by niffiwan** [comment]
how do you restart `continuous-integration/appveyor/branch` anyway? :)

**Comment by asarium** [comment]
I have no idea where that one comes from so I'll just ignore it. I think AppVeyor gets confused sometimes when building branches and pull requests...

**Pull request merged**